File: app/src/main/java/id/ac/stiki/doleno/absenin/data/source/local/EventDao.java
package id.ac.stiki.doleno.absenin.data.source.local;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import id.ac.stiki.doleno.absenin.data.entity.Event;

@Dao
public interface EventDao {
    @Query("SELECT * FROM event")
    LiveData<List<Event>> getAll();

    @Query("SELECT * FROM event WHERE uid = :eventId")
    Event getById(int eventId);

    @Insert
    void insert(Event event);

    @Insert(entity = Event.class)
    void insert(List<Event> events);

    @Query("DELETE FROM event")
    void delete();
}

File: app/src/main/java/id/ac/stiki/doleno/absenin/data/source/local/EventDao.kt
package id.ac.stiki.doleno.absenin.data.source.local

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import id.ac.stiki.doleno.absenin.data.entity.Event

@Dao
interface EventDao {
    @get:Query("SELECT * FROM event")
    val all: List<Event>

    @Query("SELECT * FROM event WHERE uid = :eventId")
    fun getById(eventId: Int): Event

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(event: Event)

    @Insert(entity = Event::class, onConflict = OnConflictStrategy.REPLACE)
    fun insert(events: List<Event>)

    @Query("DELETE FROM event")
    fun delete()
}